src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2483,27 +2483,28 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
                 LearnFromNonNullTest(receiver, ref this.State);
             }
 
-            var accessExpressionType = VisitRvalueWithState(node.AccessExpression);
+            var accessTypeWithAnnotations = VisitLvalueWithAnnotations(node.AccessExpression);
```

Comment:
>VisitLvalueWithAnnotations [](start = 44, length = 26)

Why is this correct? The left-side of a conditional access expression is still an rvalue.

Reply:
The is visiting the right-hand-side of the ?. expression.  The result is not the final result of the whole expression.  We get the lvalue of the right-hand-side (as a type with annotations) so we can use it as a type argument to `System.Nullable` if necessary.

---
In reply to: [278710760](https://github.com/dotnet/roslyn/pull/34973#discussion_r278710760) [](ancestors = 278710760)